Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,465 @@
+
+    # CVE fix versions
+    cfvs = list()
+
+    # Request latest version
+    cve_list = requests.get("https://kubernetes.io/docs/reference/issues-security/official-cve-feed/index.json",
+                            headers={"Accept": "application/json"}).json()
+
+    async with aiohttp.ClientSession() as session:
+        tasks = []
+        for cve in cve_list['items']:
+            cveid = cve['external_url'].split("=")[-1]
+            tasks.append(request_cve_data(session=session, cveid=cveid))
+        # asyncio.gather() will wait on the entire task set to be
+        # completed.  If you want to process results greedily as they come in,
+        # loop over asyncio.as_completed()
+        cve_data_list = await asyncio.gather(*tasks, return_exceptions=True)
+
+    for cve_data in cve_data_list:
+        try:
+            cve_cna = cve_data['containers']['cna']
+            cve_metrics = cve_cna['metrics']
+            cve_affected = cve_cna['affected']
+            # This data is extracted like this due to the location of the relevant information in the schema.
+            # mitre.org uses CVE schema 5.0, which is described in the link below
+            # https://github.com/CVEProject/cve-schema/tree/master/schema/v5.0
+            # The containers -> cna path contains vulnerability information like severity, which is documented
+            # under the metrics list.
+        except KeyError as e:
+            logger.debug(f"They key {e} couldn't be found in the CVE json data for CVE {cveid}.")
+            continue
+
+        is_high_severity = any(
+            re.search(r'[cC][vV][sS]{1,2}V\d', metric_key) and metric_value['baseScore'] >= CVE_SEVERITY
+            for cve_metric in cve_metrics
+            for metric_key, metric_value in cve_metric.items()
+        )
+
+        if is_high_severity:
+            affected_kubernetes_versions = [
+                parse_cve_version_information(version_info)
+                for aff in cve_affected
+                if aff['product'] == "Kubernetes"
+                for version_info in aff['versions']
+                if version_info['status'] == "affected"
+            ]
+            for cvev in affected_kubernetes_versions:
+                if cvev not in cfvs:
+                    cfvs.append(cvev)
+
+    return cfvs
+
+
+async def get_k8s_cluster_version(kubeconfig):
+    """Get the k8s version of the cluster under test."""
+    cluster_config = await kubernetes_asyncio.config.load_kube_config(kubeconfig)
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        version_api = kubernetes_asyncio.client.VersionApi(api)
+        ret = await version_api.get_code()
+
+        version = K8sVersionInfo.extract_version(ret.git_version.strip("v"))
+        version.date = datetime.datetime.strptime(ret.build_date, '%Y-%m-%dT%H:%M:%SZ')
+
+        return version, cluster_config.current_context['name']
+
+
+def is_current_version(version, k8s_version_list, cve_version_list=[]):
```

Comment:
`cve_version_list=[]` default argument value should not be mutable

Reply:
Changed it to use a non-mutable type and set an empty list later.